AirCasting/ABConnector/SDSyncController.swift

```diff
@@ -115,10 +111,20 @@ class SDSyncController {
         }
     }
     
-    private func process(fixedSessionFile: URL, deviceID: String, completion: @escaping (Bool) -> Void) throws -> [SessionUUID] {
-        let csvSession = try CSVStreamsWithMeasurements(fileURL: fixedSessionFile)
-        fixedSessionsSaver.processAndSync(csvSession: csvSession, deviceID: deviceID, completion: completion)
-        return csvSession.sessions.map { $0.uuid }
+    private func process(fixedSessionFile: URL, deviceID: String, completion: @escaping (Bool) -> Void) -> [SessionUUID]? {
+        var uploadedSessions: [SessionUUID]? = nil
+        Log.info("Processing fixed file")
+        fixedSessionsUploader.processAndUpload(fileURL: fixedSessionFile, deviceID: deviceID) { result in
+            switch result {
+            case .success(let sessions):
+                uploadedSessions = sessions
+                completion(true)
+            case .failure(let error):
+                Log.error("[SD Sync] Failed to upload sessions to backend: \(error.localizedDescription)")
+                completion(false)
+            }
+        }
+        return uploadedSessions
```

Comment:
isn't processAnUpload asynchronous tho? It mens this will always return `nil`. You need to put this into completion rather than standard return.

On that note we would be able to start using the new async/await sweetness now since it's supported for iOS13+.